### PR TITLE
fix (hatcheries,worker): book & take care on workflow

### PR DESCRIPTION
**Pull request by yesnault**
A worker can be launch to run a Pipeline Build Job or a Workflow Job
Before this commit, we can see:

```
| 2017-11-05 11:33:40 | warn | GET     | 404  | /queue/8820/infos      getPipelineBuildJobHandler> Unable to load pipeline build job id: Pipeline build not found
| 2017-11-05 11:33:40 | warn | GET     | 404  | /queue/8821/infos      getPipelineBuildJobHandler> Unable to load pipeline build job id: Pipeline build not found
| 2017-11-05 11:33:40 | warn | GET     | 404  | /queue/8812/infos      getPipelineBuildJobHandler> Unable to load pipeline build job id: Pipeline build not found
| 2017-11-05 11:33:40 | warn | GET     | 404  | /queue/8811/infos      getPipelineBuildJobHandler> Unable to load pipeline build job id: Pipeline build not found
```
from Worker launched to work on workflow job

Two separate arguments on the worker to avoid any confusion.

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: sdk/hatchery/hatchery.go [hunk 1]
```diff
@@ -68,7 +67,7 @@ type CommonConfiguration struct {
 // ID returns hatchery id
 type Interface interface {
 	Init() error
-	SpawnWorker(model *sdk.Model, jobID int64, requirements []sdk.Requirement, registerOnly bool, logInfo string) (string, error)
+	SpawnWorker(model *sdk.Model, isWorkflowJob bool, jobID int64, requirements []sdk.Requirement, registerOnly bool, logInfo string) (string, error)
```
Comment (fsamin) [inline sdk/hatchery/hatchery.go]: humm don`'t you think there is too much arguments in this function ?
passing all spawn options through a struct could be nice

File: engine/hatchery/swarm/swarm.go [hunk 2]
```diff
@@ -320,7 +320,11 @@ func (h *HatcherySwarm) SpawnWorker(model *sdk.Model, jobID int64, requirements
 	}
 
 	if jobID > 0 {
-		env = append(env, "CDS_BOOKED_JOB_ID"+"="+strconv.FormatInt(jobID, 10))
+		if isWorkflowJob {
+			env = append(env, "CDS_BOOKED_WORKFLOW_JOB_ID"+"="+strconv.FormatInt(jobID, 10))
```
Comment (fsamin) [inline engine/hatchery/swarm/swarm.go]: i would prefer `fmt.Sprinf("CDS_BOOKED_WORKFLOW_JOB_ID=%d", jobID)`

Comment (yesnault) [reply engine/hatchery/swarm/swarm.go]: done

File: sdk/hatchery/hatchery.go (hunk 1 above)
Comment (yesnault) [reply sdk/hatchery/hatchery.go]: done

**Comment by yesnault** [comment]
UT / IT / ITP OK

File: sdk/hatchery/hatchery.go [hunk 3]
```diff
@@ -54,7 +54,16 @@ type CommonConfiguration struct {
 	} `toml:"logOptions" comment:"Hatchery Log Configuration"`
 }
 
-// Interface describe an interface for each hatchery mode (mesos, local)
+// SpawnArguments contains arguments to func SpawnWorker
+type SpawnArguments struct {
+	Model         *sdk.Model
```
Comment (fsamin) [inline sdk/hatchery/hatchery.go]: no pointer

**Pull request merged**